internal/agent/device/resource/common.go

```diff
@@ -21,7 +22,7 @@ func updateMonitor(
 		return false, err
 	}
 
-	newSamplingInterval, err := time.ParseDuration(spec.ResourceMonitorSpec.SamplingInterval)
+	newSamplingInterval, err := util.ExtendedParseDuration(spec.ResourceMonitorSpec.SamplingInterval)
```

Comment:
few questions:
- what does the `SamplingInterval` represent in the context of a resource monitor. then what would be the usecase for it to be defined in days or weeks.
- where does this value come from?